### PR TITLE
Optimize postgres runner for db tests

### PR DESCRIPTION
**Pull request by aoldershaw**
[imperative mood]: https://chris.beams.io/posts/git-commit/#imperative

## What does this PR accomplish?

Make tests that require running a local Postgres server faster (~2x faster on my machine)

```
# Before changes
ginkgo -p ./atc/db  560.68s user 1597.73s system 754% cpu 4:46.20 total

# After changes
ginkgo -p ./atc/db  172.06s user 645.53s system 777% cpu 1:45.15 total
```

On CI, it seems to be a *bit* faster, but not as drastic - the DB suite went from ~3 minutes to ~2 minutes

## Changes proposed by this PR:

* Create a `TEMPLATE DATABASE` once after migrations were run and replace the `testdb` after each test with a copy of the template
  * `postgresRunner.Truncate` avoids needing to re-run migrations, but surprisingly to me, turned out to be pretty slow - it's a (relatively) constant time operation (regardless of the size of the tables), but for such small tables, it seems to be faster to just `DROP` the whole database and recreate it
  * Using a template also avoids needing to re-run migrations
* Add some minor optimizations based on the fact that durability is not a requirement (see https://www.postgresql.org/docs/13/non-durability.html)

## Contributor Checklist

- [ ] Followed [Code of conduct], [Contributing Guide] & avoided [Anti-patterns]
- [ ] [Signed] all commits
- [ ] Added tests (Unit and/or Integration)
- [ ] Updated [Documentation]
- [ ] Added release note (Optional)

[Code of Conduct]: https://github.com/concourse/concourse/blob/master/CODE_OF_CONDUCT.md
[Contributing Guide]: https://github.com/concourse/concourse/blob/master/CONTRIBUTING.md
[Anti-patterns]: https://github.com/concourse/concourse/wiki/Anti-Patterns
[Signed]: https://help.github.com/en/github/authenticating-to-github/signing-commits
[Documentation]: https://github.com/concourse/docs

## Reviewer Checklist

- [ ] Code reviewed
- [ ] Tests reviewed
- [ ] Documentation reviewed
- [ ] Release notes reviewed
- [ ] PR acceptance performed
- [ ] New config flags added? Ensure that they are added to the
  [BOSH](https://github.com/concourse/concourse-bosh-release) and
  [Helm](https://github.com/concourse/helm) packaging; otherwise, ignored for
  the [integration
  tests](https://github.com/concourse/ci/tree/master/tasks/scripts/check-distribution-env)
  (for example, if they are Garden configs that are not displayed in the
  `--help` text).

### CODE REVIEW

**Comment by aoldershaw** [comment]
Interestingly, it seems to make things slightly *slower* on CI - the DB suite took `3m42s` on this PR, whereas on previous PRs it took ~`3m10s`.

I wonder why it's so much faster on my laptop with these changes. Could be a result of some optimizations in `DROP DATABASE` or `CREATE DATABASE ... TEMPLATE ...` between Postgres versions (my laptop is running Postgres 13, while the unit image is using Postgres 11.

**Comment by aoldershaw** [comment]
> I wonder why it's so much faster on my laptop with these changes

I didn't realize we use a tmpfs on CI for postgres - it's possible that with a tmpfs the truncates are actually pretty fast, whereas on a SSD it's faster to DROP+CREATE.

It could also have been a fluke that it was slower - I hijacked into the container and reran the DB suite and it ran in just `2m54s` - not much faster, but at least not slower. I also tried bumping postgres to 13, and it ran in `2m29s` - so maybe we should bump postgres in the unit image (I don't think there's any particular reason to test against Postgres 11 in unit)

**Comment by aoldershaw** [comment]
I upgraded the Postgres version to 13 on the unit image, and the DB suite ran in `2m9s`, about 1 minute faster than the typical duration before bumping Postgres and these optimizations. I ran another PR with Postgres 13 (without the optimizations), and it's still around 3 minutes, so there's at least some improvement on CI!

File: atc/integration/integration_suite_test.go [hunk 1]
```diff
@@ -98,10 +104,11 @@ var _ = AfterEach(func() {
 	Expect(err).NotTo(HaveOccurred())
 
 	postgresRunner.DropTestDB()
+})
 
+var _ = AfterSuite(func() {
 	dbProcess.Signal(os.Interrupt)
-	err = <-dbProcess.Wait()
-	Expect(err).NotTo(HaveOccurred())
+	<-dbProcess.Wait()
```
Comment (xtremerui) [inline atc/integration/integration_suite_test.go:111]: Is it a concern if the DB process can't be ended properly? I can see if the test is ran in a container in the pipeline it should be ok. How about running the tests locally? 

It seems some other suits are doing it this way
https://github.com/concourse/concourse/blob/3dc0c519c47f6f8017c81ea7ce97520e3b27c9f8/atc/db/migration/migration_suite_test.go#L39-L41

should them be the same?

Comment (aoldershaw) [reply atc/integration/integration_suite_test.go]: I went ahead and added some helpers to the `postgresrunner` package for common Ginkgo setup/teardown logic (and used the teardown with a timeout, so now all suites running postgres will timeout after 10 seconds). Lemme know what you think

**Review by xtremerui — APPROVED** [review]
LGTM!

**Pull request merged**